api/app/src/mappings/google/models/activity.ts

```diff
@@ -1,16 +1,145 @@
-import { z } from "zod"
+import { z } from "zod";
 import { googleResp } from ".";
 
-export const sourceIdCalories = "derived:com.google.calories.expended:com.google.android.gms:merge_calories_expended";
-export const sourceIdSteps = "derived:com.google.step_count.delta:com.google.android.gms:estimated_steps";
+export const sourceIdActivity =
+  "derived:com.google.activity.segment:com.google.android.gms:merge_activity_segments";
+export const sourceIdActiveMinutes =
+  "derived:com.google.active_minutes:com.google.android.gms:merge_active_minutes";
+export const sourceIdDistance =
+  "derived:com.google.distance.delta:com.google.android.gms:merge_distance_delta";
+export const sourceIdCalories =
+  "derived:com.google.calories.expended:com.google.android.gms:merge_calories_expended";
+export const sourceIdSteps =
+  "derived:com.google.step_count.delta:com.google.android.gms:estimated_steps";
+export const sourceIdSpeed = "derived:com.google.speed:com.google.android.gms:merge_speed";
 
 export const googleActivityDataSourceId = z.enum([
+  sourceIdActivity,
+  sourceIdActiveMinutes,
   sourceIdCalories,
+  sourceIdDistance,
   sourceIdSteps,
+  sourceIdSpeed,
 ]);
 
-export const googleActivityResp = googleResp(googleActivityDataSourceId)
+export const googleActivityResp = googleResp(googleActivityDataSourceId);
 
 export type GoogleActivity = z.infer<typeof googleActivityResp>;
 export type GoogleActivityDataSourceId = z.infer<typeof googleActivityDataSourceId>;
 
+export const GoogleActivityTypes: { [index: number]: string } = {
```

Comment:
Defs NABD at this point, but wondering if it would make it easier to work with this file if we moved these constants to a separated `constants.ts` file.